src/main/kotlin/carracing/CarRace.kt

```diff
@@ -22,7 +22,8 @@ class CarRace(val cars: List<Car>, val numRounds: Int) {
 
     fun getWinners(): List<Car> {
         val maxPosition = cars.maxOf { it.position }
-        return cars.filter { it.position == maxPosition }
+        cars.forEach { it.setAsWinner(maxPosition) }
+        return cars.filter { it.isWinner }
```

Comment:
This version with `setAsWinner()` makes sense in terms of marking the winner.
However do you think we might be adding more state management than necessary here?
What if the `Car` just told us whether it's at the winning position, rather than us updating it?

```
return cars.filter { it.isPositionAt(maxPosition) }
```